### PR TITLE
Base serializers

### PR DESCRIPTION
**Pull request by imsickofmaps**

### CODE REVIEW

**Comment by imsickofmaps** [comment]
@gsvr review please

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+language: python
+python:
+  - "2.7"
+  - "3.4"
+addons:
+  postgresql: "9.4"
```
Comment (gsvr) [inline .travis.yml:6]: 9.5?

File: LICENCE [hunk 2]
```diff
@@ -0,0 +1,28 @@
+Copyright (c) 2016, Western Cape Labs (Pty) Ltd.
```
Comment (gsvr) [inline LICENCE]: Praekelt

File: scheduler/tasks.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+from celery.task import Task
+from celery.utils.log import get_task_logger
+from django.conf import settings
+
+
+logger = get_task_logger(__name__)
+
+
+class The_Incr(Task):
```
Comment (gsvr) [inline scheduler/tasks.py:12]: Remove this?

File: scheduler/views.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    """
+    permission_classes = (IsAuthenticated,)
+    queryset = Group.objects.all()
+    serializer_class = GroupSerializer
+
+
+class ScheduleViewSet(viewsets.ModelViewSet):
+
+    """
+    API endpoint that allows dummy models to be viewed or edited.
```
Comment (gsvr) [inline scheduler/views.py]: dummy models

File: scheduler/models.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        IntervalSchedule,
+        on_delete=models.CASCADE,
+        null=True
+    )
+    endpoint = models.CharField(max_length=500, null=False)
+    payload = JSONField(null=False, blank=False, default={})
+    next_send_at = models.DateTimeField(null=True, blank=True)
+    updated_at = models.DateTimeField(auto_now=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
```
Comment (gsvr) [inline scheduler/models.py]: duplication

File: scheduler/models.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+@receiver(pre_save, sender=Schedule)
+def schedule_saved(sender, instance, **kwargs):
+    if instance.cron_definition is not None and \
+            instance.celery_cron_definition is not None:
+        # clean up old cron schedule
+        instance.celery_cron_definition.delete()
+    if instance.interval_definition is not None and \
+            instance.celery_interval_definition is not None:
+        # clean up old interval schedule
```
Comment (gsvr) [inline scheduler/models.py]: Can you add a bit more explanation to the comment here? And is it supposed to refer to 'old celery interval'?  Same thing 4 lines above.

File: scheduler/models.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+    if instance.interval_definition is not None and \
+            instance.celery_interval_definition is None:
+        every, period = instance.interval_definition.split()
+        interval = {
+            "every": int(every),
+            "period": period
+        }
+        intsch = IntervalSchedule.objects.create(**interval)
+        instance.celery_interval_definition = intsch
+        # instance.save()
```
Comment (gsvr) [inline scheduler/models.py]: Remove?

File: scheduler/models.py [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+        schedule = {
+            "minute": entry.matchers.minute.input,
+            "hour": entry.matchers.hour.input,
+            "day_of_week": entry.matchers.weekday.input,
+            "day_of_month": entry.matchers.day.input,
+            "month_of_year": entry.matchers.month.input
+        }
+        cs = CrontabSchedule.objects.create(**schedule)
+        instance.celery_cron_definition = cs
+        # instance.save()
```
Comment (gsvr) [inline scheduler/models.py]: Remove?

**Comment by gsvr** [comment]
Some small changes, then :+1:

File: seed_scheduler/settings.py [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+    'DEFAULT_PERMISSION_CLASSES': (
+        'rest_framework.permissions.IsAuthenticated',
+    ),
+    'DEFAULT_FILTER_BACKENDS': ('rest_framework.filters.DjangoFilterBackend',)
+}
+
+# Webhook event definition
+HOOK_EVENTS = {
+    # 'any.event.name': 'App.Model.Action' (created/updated/deleted)
+    'schedule.added': 'scheduler.Schedule.created+'
```
Comment (KaitCrawford) [inline seed_scheduler/settings.py:134]: @imsickofmaps sorry for the ping on the closed PR but what does the '+' at the end of this action mean?

Comment (imsickofmaps) [inline seed_scheduler/settings.py:134]: @KaitCrawford  If you want a Hook to be triggered for all users, add '+' to built-in Hooks or pass user_override=False for custom_hook events

**Pull request merged**